test/performance/perf_test.py

```diff
@@ -431,13 +431,10 @@ def insert_json_file_with_file_id(filename):
 def read_json_file(filename):
     assert proc_client is not None
     coll = proc_client.perftest.corpus
-    temp = tempfile.TemporaryFile(mode="w")
-    try:
-        temp.writelines(
-            [json.dumps(doc) + "\n" for doc in coll.find({"file": filename}, {"_id": False})]
-        )
-    finally:
-        temp.close()
+    with tempfile.TemporaryFile(mode="w") as temp:
+        for doc in coll.find({"file": filename}, {"_id": False}):
+            temp.write(json.dumps(doc))
+            temp.write("\n")
```

Comment:
I rewrote this to avoid the memory bloat of keeping all the json-encoded documents alive at once. I don't think it changed the benchmark much but figure I'll keep it anyway. Note that `json.dumps(doc) + "\n"` is also bad because it reallocates a very large string.